### PR TITLE
Clarify how to write a commit message and CHANGELOGs

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

File: contributing/code/conventions.rst [hunk 1]
```diff
@@ -7,8 +7,8 @@ coding standards and conventions used in the core framework to make it more
 consistent and predictable. You are encouraged to follow them in your own
 code, but you don't need to.
 
-Method Names
-------------
+Naming a Method
+---------------
```
Comment (OskarStark) [inline contributing/code/conventions.rst:13]:
```suggestion
.. _method-names:

Naming a Method
---------------
```

to keep a deeplink working

File: contributing/code/conventions.rst [hunk 2]
```diff
@@ -77,19 +77,63 @@ must be used instead (where ``XXX`` is the name of the related thing):
...
 
+Writing a CHANGELOG Entry
+-------------------------
+
+When adding a new feature in a minor version or deprecating an existing
+behavior, an entry to the relevant CHANGELOG(s) should be added.
+
+New features and deprecations must be described in a file named
+``CHANGELOG.md`` that should be at the root directory of the modified
+Component/Bridge/Bundle.
```
Comment (OskarStark) [inline contributing/code/conventions.rst]:
```suggestion
Component, Bridge or Bundle.
```

File: contributing/code/conventions.rst (hunk 1 above)
Comment (fabpot) [reply contributing/code/conventions.rst:13]: Good point, fixed

File: contributing/code/conventions.rst [hunk 3]
```diff
@@ -77,19 +77,63 @@ must be used instead (where ``XXX`` is the name of the related thing):
...
+    5.3
+    ---
+
+     * Add `MagicConfig` that allows configuring things
+
+.. note::
+
+    The main ``CHANGELOG-*`` files at the ``symfony/symfony`` root directory
+    are automatically generated when releases are prepared and should never be
+    modified "manually".
```
Comment (OskarStark) [inline contributing/code/conventions.rst]:
```suggestion
    modified manually.
```

Comment (fabpot) [reply contributing/code/conventions.rst]: Done

File: contributing/code/conventions.rst (hunk 2 above)
Comment (fabpot) [reply contributing/code/conventions.rst]: Done

**Comment by OskarStark** [comment]
@Nyholm do you think this can be achieved by fabbot? Otherwise I am afraid, we cannot think of it all the time

**Comment by fabpot** [comment]
I'm the one managing fabbot :)

**Comment by OskarStark** [comment]
> I'm the one managing fabbot :)

Indeed, he is actively working on gh and carsonbot 😃

**Review by OskarStark — APPROVED** [review]

**Review by Nyholm — APPROVED** [review]

**Comment by dunglas** [comment]
Regarding commit messages, are you aware of the [conventional commits](https://www.conventionalcommits.org/) spec? It has a nice syntax and some tools already implement it ([GitHub bots checking message formats](https://github.com/zeke/semantic-pull-requests), [changelog generators](https://github.com/conventional-changelog/standard-version), linters, [IDE plugins](https://plugins.jetbrains.com/plugin/13389-conventional-commit) etc).

For Symfony commit messages following this spec would look like:

```
feat(serializer): add support for foo
```

```
fix(dependency-injection): add missing type hint for bar
```

```
fix(cache)!: remove function foo()

BREAKING CHANGE: function foo() has been removed to fix a security issue
```

**Comment by fabpot** [comment]
TBH, that's looks really ugly :) And we already have tools built around our conventions.

**Review by xabbuh — APPROVED** [review]

**Comment by javiereguiluz** [comment]
Thank you Fabien.

**Pull request merged**